help_to_heat/frontdoor/routing/backwards_routing.py

```diff
@@ -14,20 +15,13 @@
 
 
 def get_prev_page(current_page, answers):
-    route_page = start_page
-    route = deque([govuk_start_page])
-
-    while len(route) < max_route_length:
-        if route_page == current_page:
-            return route.pop()
-
-        if route_page == unknown_page:
-            return unknown_page
-
-        next_page = get_next_page(route_page, answers)
-
-        route.append(route_page)
+    # there is no previous page to this
+    if current_page == govuk_start_page:
+        return unknown_page
 
-        route_page = next_page
+    try:
+        route = get_route(answers, current_page)
 
-    return unknown_page
+        return route[-2]
+    except CouldNotCalculateRouteException:
+        return unknown_page
```

Comment:
Does this get automatically logged?

Reply:
no - if I remove the catch it would be

Reply:
Worth logging something in that case then: sort of thing we'd want to catch if it's happening

Reply:
probably easiest to remove the try entirely - the error would be logged by our global error handling